addons/gamification/tests/test_challenge.py

```diff
@@ -137,6 +139,57 @@ def test_20_update_all_goals_filter(self):
             "Only portal user last logged in before last challenge update should not have been updated.",
         )
 
+    def test_30_close_goals(self):
+        Goals = self.env['gamification.goal']
+
+        admin_user = self.env.ref('base.user_admin')
+        demo_user = self.robot
+
+        definition = self.env['gamification.goal.definition'].create({
+            'name': 'Dummy Goal Definition',
+            'computation_mode': 'manually',
+        })
+
+        # Create a challenge in the past
+        challenge = self.env['gamification.challenge'].create({
+            'name': 'Dummy Challenge',
+            'period': 'once',
+            'visibility_mode': 'personal',
+            'report_message_frequency': 'never',
+            'user_domain': [('id', 'in', [self.robot.id, admin_user.id, demo_user.id])],
+            'state': 'inprogress',
+            'start_date': '2024-04-29',
+            'end_date': '2024-04-30',
+            'line_ids': [(Command.CREATE, 0, {'target_goal': 4, 'definition_id': definition.id})]
+        })
+
+        # Generate the goals
+        with freeze_time('2024-04-29 00:00:00'):
+            challenge.action_check()
+
+        goals = Goals.search([('challenge_id', '=', challenge.id)])
+        admin_goal = goals.filtered(lambda x: x.user_id == admin_user)
+        demo_goal = goals.filtered(lambda x: x.user_id == demo_user)
+        robot_goal = goals.filtered(lambda x: x.user_id == self.robot)
+
+        # Set the current values
+        admin_goal.current = 4  # Reached target
+        demo_goal.current = 2  # Failed target
+        robot_goal.current = 0  # Failed target
```

Comment:
You can do this for conciseness and self-explanation, (fewer comments will be necessary below).

```suggestion
        goals = Goals.search([('challenge_id', '=', challenge.id)])
        goals[0].current = 4  # Reached target
        goals[1].current = 2  # Failed target
        goal_reached, goal_failed = goals
```